reference-architectures/cloud_deploy_flow/functions.tf

```
@@ -0,0 +1,203 @@
+resource "google_storage_bucket" "function_bucket" {
+  name = "${var.project_id}-gcf-source"
+  location = "US"
+  uniform_bucket_level_access = true 
+}
+
+data "archive_file" "createRelease" {
+  type = "zip"
+  output_path = "/tmp/function-createRelease.zip"
+  source_dir = "CloudFunctions/createRelease/"
+}
+
+data "archive_file" "cloudDeployInteractions" {
+  type = "zip"
+  output_path = "/tmp/function-cloudDeployInteractions.zip"
+  source_dir = "CloudFunctions/cloudDeployInteractions/"
+}
+
+data "archive_file" "cloudDeployOperations" {
+  type = "zip"
+  output_path = "/tmp/function-cloudDeployOperations.zip"
+  source_dir = "CloudFunctions/cloudDeployOperations/"
+}
+
+data "archive_file" "cloudDeployApprovals" {
+  type = "zip"
+  output_path = "/tmp/function-cloudDeployApprovals.zip"
+  source_dir = "CloudFunctions/cloudDeployApprovals/"
+}
+
+resource "google_storage_bucket_object" "createRelease" {
+  name = "function-create-release.zip"
+  bucket = google_storage_bucket.function_bucket.name
+  source = data.archive_file.createRelease.output_path
+}
+
+resource "google_storage_bucket_object" "cloudDeployInteractions" {
+  name = "function-cloudDeployInteractions.zip"
+  bucket = google_storage_bucket.function_bucket.name
+  source = data.archive_file.cloudDeployInteractions.output_path
+}
+
+resource "google_storage_bucket_object" "cloudDeployOperations" {
+  name = "function-cloudDeployOperations.zip"
+  bucket = google_storage_bucket.function_bucket.name
+  source = data.archive_file.cloudDeployOperations.output_path
+}
+
+resource "google_storage_bucket_object" "cloudDeployApprovals" {
+  name = "function-cloudDeployApprovals.zip"
+  bucket = google_storage_bucket.function_bucket.name
+  source = data.archive_file.cloudDeployApprovals.output_path
+}
+
+# Create a Cloud Function to trigger Cloud Deploy
+resource "google_cloudfunctions2_function" "create-release" {
+  name    = "create-release"
+  project = var.project_id
+  location = var.region
+  build_config {
+    entry_point = "deployTrigger"
+    runtime     = "go122" # Or your preferred runtime
```

"Error waiting to create function: Error waiting for Creating function: Error code 3, message: Build failed with status: FAILURE. Could not build the function due to a missing permission on the build service account."

you need to explicitly set the build SA in the resource config for all functions:
"Error waiting to create function: Error waiting for Creating function: Error code 3, message: Build failed with status: FAILURE. Could not build the function due to a missing permission on the build service account."

you need to explicitly set the build SA in the resource config for all functions:

```suggestion
    runtime     = "go122" # Or your preferred runtime
    service_account = google_service_account.cloudbuild_service_account.id
```